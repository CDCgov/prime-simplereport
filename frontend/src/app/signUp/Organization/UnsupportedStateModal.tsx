import React, { useState } from "react";

import Button from "../../commonComponents/Button/Button";
import Modal from "../../commonComponents/Modal";
import Checkboxes from "../../commonComponents/Checkboxes";
import { getStateNameFromCode } from "../../utils/state";

export type UnsupportedStateModalProps = {
  showModal: boolean;
  onClose: (clearField: boolean) => void;
  state: string;
};

export const UnsupportedStateModal: React.FC<UnsupportedStateModalProps> = ({
  showModal,
  onClose,
  state,
}) => {
  const [checkboxState, setCheckboxState] = useState(false);

  return (
    <Modal
      onClose={() => onClose(true)}
      showModal={showModal}
      showClose={true}
      contentLabel={"Unsupported State"}
      containerClassName={"unsupported-state-modal"}
    >
      <Modal.Header
        styleClassNames={"margin-0 font-sans-lg line-height-sans-2"}
      >
        {getStateNameFromCode(state)} isn't connected to SimpleReport yet.
      </Modal.Header>
      <p>
        You can join the waitlist to be notified when SimpleReport is available
        in your state, or if your organization is reporting to a SimpleReport
        state, you can continue.
      </p>
      <Checkboxes
        name="acknowledge"
        legend=""
        onChange={() => {
          setCheckboxState(!checkboxState);
        }}
        boxes={[
          {
            value: "acknowledged",
            label: (
              <>
                My organization is submitting test results to a{" "}
                <a
                  href="https://simplereport.gov/using-simplereport/manage-facility-info/find-supported-jurisdictions"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  state that is connected to SimpleReport
                </a>
                .
              </>
            ),
            checked: checkboxState,
            "aria-label": "acknowledged",
          },
        ]}
      />
      <Modal.Footer styleClassNames={"prime-right-align"}>
        <Button
          variant="outline"
          onClick={() =>
            window.open(
              "https://simplereport.gov/waitlist",
              "Simple Report WaitList",
              "noopener"
            )
          }
        >
          Join waitlist
        </Button>
        {onClose && (
          <Button
            onClick={() => {
              onClose(false);
              setCheckboxState(false);
            }}
            disabled={!checkboxState}
          >
            Continue sign up
          </Button>
        )}
      </Modal.Footer>
    </Modal>
  );
};
