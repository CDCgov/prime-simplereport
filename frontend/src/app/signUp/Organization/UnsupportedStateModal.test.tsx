import { render, RenderResult, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { UnsupportedStateModal } from "./UnsupportedStateModal";

describe("UnsupportedStateModal", () => {
  let mockOnClose = jest.fn();
  let component: RenderResult;

  beforeEach(() => {
    component = render(
      <UnsupportedStateModal
        showModal={true}
        onClose={mockOnClose}
        state={"MI"}
      />
    );
  });
  it("renders", () => {
    expect(component).toMatchSnapshot();
  });

  it("closes on continue", () => {
    const continueButton = expect(continueButton).toBeDisabled();

    screen.getByLabelText("acknowledged").click();
    expect(continueButton).toBeEnabled();
    continueButton.click();

    expect(mockOnClose).toBeCalledWith(false);
  });

  it("closes on esc key", async () => {
    expect(mockOnClose).not.toHaveBeenCalled();
    await userEvent.keyboard("{Escape}");
    expect(mockOnClose).toHaveBeenCalledWith(true);
  });
});
