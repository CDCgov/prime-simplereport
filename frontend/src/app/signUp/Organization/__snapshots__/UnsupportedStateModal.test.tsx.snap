// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnsupportedStateModal renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div />
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div
      class="modal--basic"
    >
      <div
        class="ReactModal__Overlay prime-modal-overlay display-flex flex-align-center flex-justify-center"
      >
        <div
          aria-label="Unsupported State"
          aria-modal="true"
          class="ReactModal__Content"
          role="dialog"
          style="position: initial; top: 40px; left: 40px; right: 40px; bottom: 40px; border: 0px; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 0px;"
          tabindex="-1"
        >
          <div
            class="modal__container unsupported-state-modal"
          >
            <button
              class="modal__close-button"
              style="cursor: pointer; z-index: 1;"
            >
              <img
                alt="Close"
                class="modal__close-img"
                src="close.svg"
              />
            </button>
            <main
              class="padding-205"
            >
              <div
                class="modal__content grid-row usa-prose"
              >
                <div
                  class="grid-col"
                >
                  <h1
                    class="modal__heading margin-0 font-sans-lg line-height-sans-2"
                  >
                    Michigan
                     isn't connected to SimpleReport yet.
                  </h1>
                  <p>
                    You can join the waitlist to be notified when SimpleReport is available in your state, or if your organization is reporting to a SimpleReport state, you can continue.
                  </p>
                  <div
                    class="usa-form-group"
                  >
                    <fieldset
                      class="usa-fieldset prime-checkboxes"
                    >
                      <legend
                        class="usa-legend"
                      />
                      <div
                        class="grid-row checkboxes"
                      >
                        <div
                          class="tablet:grid-col"
                        >
                          <div
                            class="usa-checkbox"
                          >
                            <input
                              aria-label="acknowledged"
                              class="usa-checkbox__input"
                              data-testid="acknowledge-acknowledged"
                              id="acknowledge-acknowledged"
                              name="acknowledge"
                              type="checkbox"
                              value="acknowledged"
                            />
                            <label
                              class="usa-checkbox__label"
                              for="acknowledge-acknowledged"
                            >
                              My organization is submitting test results to a
                               
                              <a
                                href="https://simplereport.gov/using-simplereport/manage-facility-info/find-supported-jurisdictions"
                                rel="noopener noreferrer"
                                target="_blank"
                              >
                                state that is connected to SimpleReport
                              </a>
                              .
                            </label>
                          </div>
                        </div>
                      </div>
                    </fieldset>
                  </div>
                  <div
                    class="modal__footer prime-right-align"
                  >
                    <button
                      class="usa-button usa-button--outline"
                      type="button"
                    >
                      Join waitlist
                    </button>
                    <button
                      aria-disabled="true"
                      class="usa-button usa-button-disabled"
                      disabled=""
                      type="button"
                    >
                      Continue sign up
                    </button>
                  </div>
                </div>
              </div>
            </main>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
